### ISSUE TITLE
Questions for using spdx_tools with V3 Model

### ISSUE THREAD

**Issue by willarmiros** [spdx/tools-python#680]
Hello! I am trying to create SBOMs programmatically using the `prototype-spdx-3.0` branch of this repo. It's been really great to use so far and I completely understand that it is unreleased, so not everything is complete yet. I had some questions on best practices when using the repo below.

Happy to open PRs to help if any of these turn out to be bugs, just looking for guidance on that front!

1. **What is the recommended way to generate SPDX IDs for elements, and are there any restrictions on them?** I would expect to just create a UUID, but the [README example](https://github.com/spdx/tools-python#example) uses very human-readable strings, so I wasn't sure which to use on my end.
2. **Is creationInfo required on every element?** I initially assumed it was not since I felt that would require _alot_ of repetitive/empty fields. However when I try to create an element, e.g. a Relationship, without it then serialize with the debug writer I get this error (which is resolved when I add CreationInfo):

```
  File "/Users/williamarmiros/bom/spdx.py", line 97, in __init__
    relationship_writer.write_relationship(relationship, sys.stdout)
  File "/Users/williamarmiros/open-source/spdx/tools-python/src/spdx_tools/spdx3/writer/console/relationship_writer.py", line 14, in write_relationship
    write_element_properties(relationship, text_output)
  File "/Users/williamarmiros/open-source/spdx/tools-python/src/spdx_tools/spdx3/writer/console/element_writer.py", line 18, in write_element_properties
    write_creation_info(element.creation_info, text_output, True)
  File "/Users/williamarmiros/open-source/spdx/tools-python/src/spdx_tools/spdx3/writer/console/creation_info_writer.py", line 13, in write_creation_info
    write_value("specVersion", str(creation_info.spec_version), text_output, indent)
AttributeError: 'NoneType' object has no attribute 'spec_version'
```

3. **Which Profile(s) should I use in CreationInfo?** This is somewhat dependent on the above question, but if I have a `SpdxDocument` element, which in turn contains several other elements, should I list only the `Core` profile in the SpdxDocument's creationInfo, or should I include all the profiles of its sub-elements?

**Comment by armintaenzertng** [comment]
1. There are no restrictions on SPDX Ids that I know of, apart from them having to be globally unique. That is, if you use your personal namespace and make sure that your IDs are unique you should be fine. Of course, you can also create a UUID (which will have the downside of not being very readable, but you might not care about that).
2. That is an oversight, thanks for catching that! Fix is in #684.
3. That is a model issue and a good question that I don't know the answer to, either. If you don't mind, please open an issue about this in the [spec-repo](https://github.com/spdx/spdx-3-model).

**Comment by willarmiros** [comment]
1. That makes sense, thanks for the guidance
2. Great, so it seems like creationInfo is optional after all! 
3. Will do, thanks! 


I think this answers all my questions so I'll close out the issue.
